### PR TITLE
Fixed bug allowing null values into PropertyBag.

### LINKED ISSUE #1958: System.Reflection.TargetInvocationException after run finished
After running tests (`/packages/NUnit.ConsoleRunner.3.5.0/tools/nunit3-console.exe workspace/Tests/ED_DC_Tests.dll --where "cat == _Acceptance" --work workspace --inprocess `) there is error:

```
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at NUnit.Framework.Internal.PropertyBag.AddToXml(TNode parentNode, Boolean recursive)
   at NUnit.Framework.Internal.Test.PopulateTestNode(TNode thisNode, Boolean recursive)
   at NUnit.Framework.Internal.TestMethod.AddToXml(TNode parentNode, Boolean recursive)
   at NUnit.Framework.Internal.TestResult.AddToXml(TNode parentNode, Boolean recursive)
   at NUnit.Framework.Internal.TestResult.AddToXml(TNode parentNode, Boolean recursive)
   at NUnit.Framework.Internal.TestResult.AddToXml(TNode parentNode, Boolean recursive)
   at NUnit.Framework.Internal.TestResult.AddToXml(TNode parentNode, Boolean recursive)
   at NUnit.Framework.Internal.TestResult.AddToXml(TNode parentNode, Boolean recursive)
   at NUnit.Framework.Internal.TestResult.AddToXml(TNode parentNode, Boolean recursive)
   at NUnit.Framework.Api.FrameworkController.RunTests(ICallbackEventHandler handler, String filter)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes, StackCrawlMark& stackMark)
   at System.Activator.CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes)
   at System.Activator.CreateInstance(String assemblyString, String typeName, Boolean ignoreCase, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes, Evidence securityInfo, StackCrawlMark& stackMark)
   at System.Activator.CreateInstance(String assemblyName, String typeName, Boolean ignoreCase, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes, Evidence securityInfo)
   at System.AppDomain.CreateInstance(String assemblyName, String typeName, Boolean ignoreCase, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes, Evidence securityAttributes)
   at System.AppDomain.CreateInstanceAndUnwrap(String assemblyName, String typeName, Boolean ignoreCase, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes, Evidence securityAttributes)
   at System.AppDomain.CreateInstanceAndUnwrap(String assemblyName, String typeName, Boolean ignoreCase, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes, Evidence securityAttributes)
   at NUnit.Engine.Drivers.NUnit3FrameworkDriver.Run(ITestEventListener listener, String filter)
   at NUnit.Engine.Runners.DirectTestRunner.RunTests(ITestEventListener listener, TestFilter filter)
   at NUnit.Engine.Runners.MasterTestRunner.RunTests(ITestEventListener listener, TestFilter filter)
   at NUnit.Engine.Runners.MasterTestRunner.Run(ITestEventListener listener, TestFilter filter)
   at NUnit.ConsoleRunner.ConsoleRunner.RunTests(TestPackage package, TestFilter filter)
   at NUnit.ConsoleRunner.Program.Main(String[] args)
```

The error is in some test, because other categories runs good. All tests runs in parallel. Any suggesting to find buggy case?

### PR DESCRIPTION
**Pull request by OmicronPersei**
Closes #1958 .

### CODE REVIEW

File: src/NUnitFramework/framework/Internal/PropertyBag.cs [hunk 1]
```diff
@@ -49,13 +49,20 @@ public class PropertyBag : IPropertyBag
...
+                if (!inner.TryGetValue(key, out list))
+                {
+                    list = new List<object>();
+                    inner.Add(key, list);
+                }
+                list.Add(value);
+            }
+            else
+            {
+                throw new ArgumentException("The provided value cannot be null.");
```
Comment (CharliePoole) [inline src/NUnitFramework/framework/Internal/PropertyBag.cs]: I'd use our standard `Guard.ArgumentNotNull` for this.

**Comment by OmicronPersei** [comment]
Since we are waiting on Travis CI much more often than we hope, does Github have a feature for auto merging upon CI success?  I know GitLab had something like that.

**Comment by mikkelbu** [comment]
@OmicronPersei I've edited your first comment (removed the word "issue"), so that this PR will autoclose #1958.

**Comment by OmicronPersei** [comment]
Did not know that would prevent the autoclose of an issue!  A little surprising since GitHub officially calls them "issues" though.  Thanks!

File: src/NUnitFramework/framework/Internal/PropertyBag.cs [hunk 2]
```diff
@@ -49,6 +49,8 @@ public class PropertyBag : IPropertyBag
         /// <param name="value">The value</param>
         public void Add(string key, object value)
         {
+            Guard.ArgumentNotNull(value, "value");
```
Comment (jnm2) [inline src/NUnitFramework/framework/Internal/PropertyBag.cs]: If CI hasn't passed by the time you see this, I'd really like to have `nameof(value)` here. I've seen the top line of an NUnit method throw an exception that had an old param name in it.

**Comment by CharliePoole** [comment]
Travis has been waiting for a host for osx for almost four hours. Everything else has passed so I'm merging.

File: src/NUnitFramework/framework/Internal/PropertyBag.cs (hunk 2 above)
Comment (OmicronPersei) [reply src/NUnitFramework/framework/Internal/PropertyBag.cs]: Coincidentally I thought of doing it this way but I was dissuaded by rarely seeing it done in the rest of the codebase.  :(

Comment (jnm2) [reply src/NUnitFramework/framework/Internal/PropertyBag.cs]: Nice 🙃 @ChrisMaddock did an initial sweep to get us onto `nameof` in https://github.com/nunit/nunit/pull/2623.

**Pull request merged**